pkg/vm/vm.go

```diff
@@ -63,6 +63,15 @@ const (
 // SyscallHandler is a type for syscall handler.
 type SyscallHandler = func(*VM, uint32) error
 
+// OnExecHook is a type for a callback that is invoked
+// after each instruction is executed.
```

Comment:
Oh, it's my bad, I've reviewed one more time and it's actually executed *before* the instruction execution itself (which is correct). `ctx.Next()` itself doesn't execute the instruction, it's `v.execute(ctx, op, param)` who does.

Reply:
Oh, I totally missed that

Reply:
Done